### ISSUE TITLE
Crash on `MuxStatsExoPlayer.updateCustomerData(CustomerData)`

### ISSUE THREAD

**Issue by zilinx** [muxinc/mux-stats-sdk-exoplayer#137]
Mux SDK version `2.4.14` for ExoPlayer `2.14.1+`
ExoPlayer version `2.14.2` 

The call [`updateCustomerData(CustomerData)`](https://github.com/muxinc/mux-stats-sdk-exoplayer/blob/9568e3c20d92838ca4de2d6f9c71c5433b1986a1/MuxExoPlayer/src/main/java/com/mux/stats/sdk/muxstats/MuxBaseExoPlayer.java#L232) causes a crash when trying to request an update using the call using a new instance of `CustomerData`:
```
 Process: no.tv2.sumo.debug, PID: 12614
    java.lang.NullPointerException: Attempt to invoke virtual method 'boolean com.mux.stats.sdk.core.model.b.isEqualTo(com.mux.stats.sdk.core.model.b)' on a null object reference
        at com.mux.stats.sdk.muxstats.g.a(Unknown Source:65)
        at com.mux.stats.sdk.muxstats.e.updateCustomerData(Unknown Source:2)
        at [app's call to Mux SDK]
 ```
The crash does not happen when using any of the deprecated `updateCustomerData` methods. When using an updated version of existing CustomerData it does not crash, but the changes are not applied (not visible in the dashboard).

Code sample below. It makes no difference whether `updatedVideoData` is an entirely new instance of `CustomerVideoData` or a modified instance from `muxStats` is passed.
```
val muxStats: MuxStatsExoPlayer
// muxStats is initialised and starts tracking. The constructor with `CustomerData` is used.
// 1. using deprecated method - use current player data, replace videoData - ok.
muxStats.updateCustomerData(muxStats.customerData.customerPlayerData, updatedVideoData) 
// 2. using a modified instance of CustomerData - does not crash, but in the dashboard the title for this view is not set.
val data = muxStats.customerData
data.customerVideoData.videoTitle = "Updated Title"
muxStats.updateCustomerData(data)
// 3. using a new instance CustomerData - use current player and view data, replace videoData - crash.
muxStats.updateCustomerData(
    CustomerData(
        muxStats.customerData.customerPlayerData,
        updatedVideoData,
        muxStats.customerData.customerViewData
    )
)
```

**Comment by jsanford8** [comment]
Hey @zilinx - thanks for submitting this, we're working on a fix

**Comment by zilinx** [comment]
Has this issue been addressed? I was just testing with a variation of scenario 2 from my first post (example below), using Mux SDK version `2.5.2` for ExoPlayer 2.14.1. It no longer produces that crash.

```
val videoData = mux.customerData.customerVideoData
videoData.videoTitle = "streamData"
mux.customerData.customerVideoData = videoData
mux.updateCustomerData(mux.customerData)  
```

By the way, did you consider introducing a builder pattern? I feel that something like ExoPlayer's [MediaItem.buildUpon()](https://github.com/google/ExoPlayer/blob/02f7aafe67b4893134a31ea55a3ba0f0535df145/library/common/src/main/java/com/google/android/exoplayer2/MediaItem.java#L1238) would work really well here.

**Comment by skidder** [comment]
@zilinx thanks for the update. Yes, this specific issue has been addressed in the 2.5.2 release. We'll look into the `MediaItem.buildUpon()` pattern as well, I appreciate the suggestion!
